### PR TITLE
Add documentation, support, and lint for tentative directories

### LINKED ISSUE #23300: Document tentative directories
https://github.com/web-platform-tests/wpt/pull/7663 introduced the .tentative file name flag, and since then lots of files have begun to use this convention.

However, we also have 3 directories containing "tentative":
- [css/css-grid/masonry.tentative](https://github.com/web-platform-tests/wpt/tree/3320205a9d70636efc03cfff340edecd0bee4694/css/css-grid/masonry.tentative) added in https://github.com/web-platform-tests/wpt/pull/23286
- [custom-elements/state.tentative](https://github.com/web-platform-tests/wpt/tree/3320205a9d70636efc03cfff340edecd0bee4694/custom-elements/state.tentative) added in https://github.com/web-platform-tests/wpt/pull/21541
- [websockets/stream-tentative](https://github.com/web-platform-tests/wpt/tree/3320205a9d70636efc03cfff340edecd0bee4694/websockets/stream-tentative) added in https://github.com/web-platform-tests/wpt/pull/18335

This is obviously convenient when there are a lot of tentative tests in a group, but it's something we haven't documented, and the naming convention has already diverged.

@web-platform-tests/wpt-core-team FYI

### PR DESCRIPTION
**Pull request by stephenmcgruer**
Tentative directories were in-use already, but not documented or clearly
supported. This PR documents them, adds support for it in the only place
that cares about tentative (manifest/sourcefile.py), and adds a lint rule
to enforce the expected naming.

Directories that break the lint rule are also renamed.

Fixes https://github.com/web-platform-tests/wpt/issues/23300

### CODE REVIEW

**Comment by stephenmcgruer** [comment]
The fact that we had to rename `css-tables/tentative/` to `css-tables/tables.tentative/` (because it covers tentative issues across the whole spec) might suggest that consolidating on 'this is the only way' isn't right - wdyt @foolip ?

**Comment by jgraham** [comment]
The code changes look good to me on their own terms, but I note that seeing the changes makes me think we should have picked `tentative` as the only possible name, and just accepted `css/css-masonry/tentative/` especially if we're worried about top-level directories not matching spec shortnames (bearing in mind that `css` is a special case so `css/css-masonry.tentative` is basically the same as a top level directory with `.tentative` in its name).

**Comment by jgraham** [comment]
Another thing to consider is what the transition path looks like when things are standardised. In many cases not all features will be added to a standard at the same time, so forcing `tentative` to be a single subdirectory rather than a dotted part of some top level name provides a clear transition path; you move the files that are testing stabilised features up one level.

**Review by FremyCompany — COMMENTED** [review]
Hi, I'm fine whichever way this is going, but my quick thoughts is that I think the fixed-name `tentative` folder is a good enough without allowing for arbitrary dot patterns, because if you want a subfolder you can do `/subfolder/tentative/test.html` with potentially no test in `subfolder` directly.

**Comment by stephenmcgruer** [comment]
Ping @foolip @jgraham PTAL.

File: tools/lint/rules.py [hunk 1]
```diff
@@ -341,6 +341,12 @@ class DuplicateBasenamePath(Rule):
     to_fix = "rename files so they have unique basename paths"
 
 
+class TentativeDirectoryName(Rule):
+    name = "TENTATIVE-DIRECTORY-NAME"
+    description = "Directories for tentative files must be named exactly 'tentative'"
```
Comment (foolip) [inline tools/lint/rules.py:346]:
```suggestion
    description = "Directories for tentative tests must be named exactly 'tentative'"
```

**Review by foolip — APPROVED** [review]

**Review by FremyCompany — APPROVED** [review]

**Pull request merged**